draft-ietf-webtrans-http3.md

```diff
@@ -231,56 +232,55 @@ type, followed by the session ID, encoded as a variable-length integer, followed
 by the user-specified stream data ({{fig-unidi}}).
 
 ~~~~~~~~~~ drawing
-  0                   1                   2                   3
-  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                           0x54 (i)                          ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                        Session ID (i)                       ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-  |                         Stream Body                         ...
-  +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+Stream Type (i) = 0x54,
+Session ID (i),
+Stream Body (..),
 ~~~~~~~~~~
 {: #fig-unidi title="Unidirectional WebTransport stream format"}
 
 ## Bidirectional Streams
 
-WebTransport endpoints can initiate bidirectional streams by opening an HTTP/3
-bidirectional stream and then immediately sending a special signal value 0x41,
-followed by the associated session ID, both encoded as a variable-length
-integer; the rest of the stream is the application payload of the WebTransport
-stream ({{fig-bidi-client}}).
+All client-initiated bidirectional streams are reserved by HTTP/3 as request
+streams, which are a sequence of HTTP/3 frames with a variety of rules; see
+{{Sections 4.1 and 6.1 of HTTP3}}.
+
+WebTransport extends HTTP/3 to allow clients to declare and use alternative
+request stream rules.  Once WebTransport has been established
```

Comment:
I would prefer to use "syntax" instead of "rules" here and elsewhere.

Reply:
RFC9114 doesn't declare stream types as syntax though. And the problem is you can send things with the correct byte layout but in the wrong order, which violates the rules set out in 9114 Section 4.1. Specifically, I used `rule(s)` as a direct reference to https://www.rfc-editor.org/rfc/rfc9114.html#section-4.1-7

> Receipt of an invalid sequence of frames MUST be treated as a [connection error](https://www.rfc-editor.org/rfc/rfc9114.html#errors) of type [H3_FRAME_UNEXPECTED](https://www.rfc-editor.org/rfc/rfc9114.html#H3_FRAME_UNEXPECTED). In particular, a [DATA](https://www.rfc-editor.org/rfc/rfc9114.html#frame-data) frame before any [HEADERS](https://www.rfc-editor.org/rfc/rfc9114.html#frame-headers) frame, or a [HEADERS](https://www.rfc-editor.org/rfc/rfc9114.html#frame-headers) or [DATA](https://www.rfc-editor.org/rfc/rfc9114.html#frame-data) frame after the trailing [HEADERS](https://www.rfc-editor.org/rfc/rfc9114.html#frame-headers) frame, is considered invalid. Other frame types, especially unknown frame types, might be permitted subject to their own rules; see [Section 9](https://www.rfc-editor.org/rfc/rfc9114.html#extensions).